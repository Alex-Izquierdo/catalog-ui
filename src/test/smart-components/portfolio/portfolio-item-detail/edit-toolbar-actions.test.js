import React from 'react';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import { MemoryRouter } from 'react-router-dom';

import EditToolbarActions from '../../../../SmartComponents/Portfolio/portfolio-item-detail/edit-toolbar-actions';

describe('<EditToolbarActions />', () => {
  let initialProps;

  beforeEach(() => {
    initialProps = {
      detailUrl: '/foo/bar',
      onSave: jest.fn(),
      resetWorkflow: jest.fn()
    };
  });

  it('should render correctly', () => {
    const wrapper = mount(
      <MemoryRouter>
        <EditToolbarActions { ...initialProps } />
      </MemoryRouter>
    );
    expect(toJson(wrapper.find(EditToolbarActions))).toMatchSnapshot();
  });
});
