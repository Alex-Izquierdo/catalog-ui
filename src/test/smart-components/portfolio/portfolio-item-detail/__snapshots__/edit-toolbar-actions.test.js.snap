// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EditToolbarActions /> should render correctly 1`] = `
<EditToolbarActions
  detailUrl="/foo/bar"
  onSave={[MockFunction]}
  resetWorkflow={[MockFunction]}
>
  <LevelItem>
    <div>
      <Link
        replace={false}
        to="/foo/bar"
      >
        <a
          href="/foo/bar"
          onClick={[Function]}
        >
          <Button
            aria-label={null}
            className=""
            component="button"
            isActive={false}
            isBlock={false}
            isDisabled={false}
            isFocus={false}
            isHover={false}
            onClick={[MockFunction]}
            type="button"
            variant="link"
          >
            <button
              aria-disabled={null}
              aria-label={null}
              className="pf-c-button pf-m-link"
              disabled={false}
              onClick={[MockFunction]}
              tabIndex={null}
              type="button"
            >
              Cancel
            </button>
          </Button>
        </a>
      </Link>
    </div>
  </LevelItem>
  <LevelItem>
    <div>
      <Button
        aria-label={null}
        className=""
        component="button"
        isActive={false}
        isBlock={false}
        isDisabled={false}
        isFocus={false}
        isHover={false}
        onClick={[MockFunction]}
        type="button"
        variant="primary"
      >
        <button
          aria-disabled={null}
          aria-label={null}
          className="pf-c-button pf-m-primary"
          disabled={false}
          onClick={[MockFunction]}
          tabIndex={null}
          type="button"
        >
          Save
        </button>
      </Button>
    </div>
  </LevelItem>
</EditToolbarActions>
`;
